mmv1/products/accesscontextmanager/api.yaml

```diff
@@ -899,6 +905,220 @@ objects:
                 The list of APIs usable within the Service Perimeter.
                 Must be empty unless `enableRestriction` is True.
               item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: 'ingressPolicies'
+            description: /
+              List of `IngressPolicies` to apply to the perimeter. A perimeter may
+              have multiple `IngressPolicies`, each of which is evaluated
+              separately. Access is granted if any `Ingress Policy` grants it.
+              Must be empty for a perimeter bridge.
+            at_least_one_of:
+              - status.0.resources
+              - status.0.access_levels
+              - status.0.restricted_services
+              - status.0.ingress_policies
+              - status.0.egress_policies
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+              - !ruby/object:Api::Type::NestedObject                  
+                name: 'ingressFrom'
+                description: /
+                  Defines the conditions on the source of a request causing this `IngressPolicy`
+                  to apply.
+                at_least_one_of:
+                  - status.0.ingress_policies.ingressFrom
+                  - status.0.ingress_policies.ingressTo
+                properties:
+                - !ruby/object:Api::Type::Enum
+                  name: 'identityType'
+                  description: /
+                    Specifies the type of identities that are allowed access from outside the 
+                    perimeter. If left unspecified, then members of `identities` field will be 
+                    allowed access.
+                  values:
+                    - :IDENTITY_TYPE_UNSPECIFIED
+                    - :ANY_IDENTITY
+                    - :ANY_USER_ACCOUNT
+                    - :ANY_SERVICE_ACCOUNT
+                - !ruby/object:Api::Type::Array
+                  name: 'identities'
+                  item_type: Api::Type::String
+                  description: /
+                    A list of identities that are allowed access through this ingress policy.
+                    Should be in the format of email address. The email address should represent 
+                    individual user or service account only.
+                - !ruby/object:Api::Type::Array
+                  name: 'sources'
+                  description: /
+                    Sources that this `IngressPolicy` authorizes access from.                       
```

Comment:
Can we make this required: true also?

Reply:
I have been reviewing the API and running some tests against it. The API does not seem to have any requirement of required fields or "at least / exactly one of".

For example
- it is possible to create a blank `ingressPolicy` without an `ingressFrom` or `ingressTo`. 
- It is possible to create a blank `ingressPolicy.ingressFrom` without specifying an `identityType`, `identities`, `allowedIdentity` or `sources` . 
- An `ingressFrom` can be empty, have only `sources`, or have a `sources` + `identityType`
- There are incompatible combinations, but those should be obvious to the user. If the user still goes ahead the error is self explanatory. E.g. can't mix `identityType: ANY_IDENTITY` and at thiese same time specify users in `identities`. The error message raised is: `[...] Error in IngressFrom: 'identities' field should be empty when identity_type field is not set to IDENTITY_TYPE_UNSPECIFIED.`
- The rules seem to apply equally for `egress`

My recommendation is to remove all the `at_least_one_of` in this PR. Any other suggestions?

Reply:
Yeah, I guess that is the only way to handle this